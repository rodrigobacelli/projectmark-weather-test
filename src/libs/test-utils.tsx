import * as React from 'react';
import { render, RenderOptions } from '@testing-library/react';

const Providers = ({ children }: { children: React.ReactNode }) => {
  return children;
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { customRender as render };
