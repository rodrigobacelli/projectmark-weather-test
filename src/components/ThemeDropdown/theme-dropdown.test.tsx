import { describe, it, expect } from 'vitest';
import { render } from '../../tests/test-utils.tsx';
import { ThemeDropdown } from './';

describe('components/ThemeDropdown', () => {
  it('should handle selecting a theme', async () => {
    const result = render(
      <ThemeDropdown className="theme-dropdown-className" />
    );

    expect(result.getByRole('combobox')).toContainHTML('System');

    await result.getByRole('combobox').click();
    await result.getByText('Dark').click();

    expect(result.getByRole('combobox')).toContainHTML('Dark');
  });

  it('should accept classNames', () => {
    const result = render(
      <ThemeDropdown className="theme-dropdown-className" />
    );

    expect(
      result.container.querySelector('.theme-dropdown-className')
    ).toBeInTheDocument();
  });

  it('should render the Header Component', () => {
    const result = render(
      <ThemeDropdown className="theme-dropdown-className" />
    );

    expect(result).toMatchSnapshot();
  });
});
