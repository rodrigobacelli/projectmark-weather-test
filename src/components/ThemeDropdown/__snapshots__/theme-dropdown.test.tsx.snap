// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/ThemeDropdown > should render the Header Component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative theme-dropdown-className"
      >
        <div
          aria-controls="theme-dropdown-listbox"
          aria-expanded="false"
          aria-haspopup="listbox"
          class="hover:cursor-pointer rounded-full px-4 py-2 border border-gray-800 dark:border-gray-200 hover:bg-gray-800/20 hover:dark:bg-gray-200/20"
          id="theme-dropdown"
          role="combobox"
          tabindex="0"
        >
          <p
            class="text-gray-900 dark:text-gray-100 flex gap-2 items-center text-sm"
          >
            <svg
              class="w-4 h-4"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M223.5 32C100 32 0 132.3 0 256S100 480 223.5 480c60.6 0 115.5-24.2 155.8-63.4c5-4.9 6.3-12.5 3.1-18.7s-10.1-9.7-17-8.5c-9.8 1.7-19.8 2.6-30.1 2.6c-96.9 0-175.5-78.8-175.5-176c0-65.8 36-123.1 89.3-153.3c6.1-3.5 9.2-10.5 7.7-17.3s-7.3-11.9-14.3-12.5c-6.3-.5-12.6-.8-19-.8z"
              />
            </svg>
            <span
              class="hidden sm:inline-flex"
            >
              Light
            </span>
            <svg
              class="ml-2 w-3 h-3"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
              />
            </svg>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative theme-dropdown-className"
    >
      <div
        aria-controls="theme-dropdown-listbox"
        aria-expanded="false"
        aria-haspopup="listbox"
        class="hover:cursor-pointer rounded-full px-4 py-2 border border-gray-800 dark:border-gray-200 hover:bg-gray-800/20 hover:dark:bg-gray-200/20"
        id="theme-dropdown"
        role="combobox"
        tabindex="0"
      >
        <p
          class="text-gray-900 dark:text-gray-100 flex gap-2 items-center text-sm"
        >
          <svg
            class="w-4 h-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M223.5 32C100 32 0 132.3 0 256S100 480 223.5 480c60.6 0 115.5-24.2 155.8-63.4c5-4.9 6.3-12.5 3.1-18.7s-10.1-9.7-17-8.5c-9.8 1.7-19.8 2.6-30.1 2.6c-96.9 0-175.5-78.8-175.5-176c0-65.8 36-123.1 89.3-153.3c6.1-3.5 9.2-10.5 7.7-17.3s-7.3-11.9-14.3-12.5c-6.3-.5-12.6-.8-19-.8z"
            />
          </svg>
          <span
            class="hidden sm:inline-flex"
          >
            Light
          </span>
          <svg
            class="ml-2 w-3 h-3"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
            />
          </svg>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
